rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```
@@ -441,30 +461,111 @@ BtcBlockStoreWithCache getBtcBlockStore() {
+            tryToProcessPegin(btcTx, rskTx);
+        } else {
+            legacyProcessPegin(btcTx, rskTx, height, btcTxHash);
+        }
+    }
+
+    private void tryToProcessPegin(BtcTransaction btcTx, Transaction rskTx) throws IOException, RegisterBtcTransactionException {
+        Coin totalAmount = computeTotalAmountSent(btcTx);
+
+        PeginInformation peginInformation = new PeginInformation(
+            btcLockSenderProvider,
+            peginInstructionsProvider,
+            activations
         );
+
+        PeginEvaluationResult peginEvaluationResult = evaluatePegin(btcTx, peginInformation);
+
+        Optional<RejectedPeginReason> rejectedPeginReason = peginEvaluationResult.getRejectedPeginReason();
+        switch (peginEvaluationResult.getPeginProcessAction()) {
+            case CAN_BE_REGISTERED:
+                executePegIn(btcTx, peginInformation, totalAmount);
+                markTxAsProcessed(btcTx);
+                break;
+            case CAN_BE_REFUNDED:
+                rejectedPeginReason.ifPresent(rejectedReason -> eventLogger.logRejectedPegin(btcTx, rejectedReason));
+                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTx, totalAmount);
+                markTxAsProcessed(btcTx);
+                throw new RegisterBtcTransactionException(String.format("Error while trying to parse peg-in information for tx %s", btcTx.getHash()));
+            case CANNOT_BE_PROCESSED:
+                handleUnprocessableBtcTx(btcTx, peginInformation, rejectedPeginReason);
+                break;
+            default:
+                if (rejectedPeginReason.isPresent()){
+                    throw new RegisterBtcTransactionException(String.format("Error while trying to process pegin. Error Reason: %s", rejectedPeginReason));
+                } else {
+                    throw new RegisterBtcTransactionException(String.format("Error while trying to process pegin."));
+                }
```

This is also something that should never happen, so I'd simply throw an exception here. `evaluatePegin` should always respond with one of the above results.

Maybe the default case is that it's unprocessable?